### PR TITLE
(prometheus) show only label name in label matcher

### PR DESCRIPTION
**Pull request by mtanda**
I want to suppress obviously inappropriate candidate in label matcher.

@alexanderzobnin Please review this.

### CODE REVIEW

File: public/app/core/components/code_editor/mode-prometheus.js [hunk 1]
```diff
@@ -49,15 +49,42 @@ var PrometheusHighlightRules = function() {
...
+      regex : "[\\])]"
     }, {
       token : "paren.rparen",
-      regex : "[\\])}]"
+      regex : "}"
     }, {
       token : "text",
       regex : "\\s+"
+    } ],
+    "start-label-matcher" : [ {
```
Comment (torkelo) [inline public/app/core/components/code_editor/mode-prometheus.js]: Thanks for this @mtanda I was looking for ways to to this! you figured it out!

File: public/app/plugins/datasource/prometheus/completer.ts [hunk 2]
```diff
@@ -9,6 +9,24 @@ export class PromCompleter {
   }
 
   getCompletions(editor, session, pos, prefix, callback) {
+    let token = session.getTokenAt(pos.row, pos.column);
+
+    switch (token.type) {
+      case 'label.name':
+        callback(null, ['instance', 'job'].map(function (key) {
```
Comment (torkelo) [inline public/app/plugins/datasource/prometheus/completer.ts]: would be cool to parse out the preceding metric name and lookup label values here :)

Comment (mtanda) [reply public/app/plugins/datasource/prometheus/completer.ts]: Yeah, I want to do it.
I will try it in this weekend :)

**Comment by mtanda** [comment]
In 4.5 release, this change doesn't work correctly.

By token name change,
https://github.com/grafana/grafana/commit/e5f84484eb1210d835c06f4f60229abc3b05d9a8

It becomes unexpected token name in this line.
https://github.com/grafana/grafana/blob/v4.5.0/public/app/plugins/datasource/prometheus/mode-prometheus.js#L404

**Comment by torkelo** [comment]
does it result in an error?

**Comment by torkelo** [comment]
looks fine when I test (no errors), I did not care that this was issue is not working as there was so many prometheus changes in new PRs and I did not want to merge those before release

**Comment by torkelo** [comment]
I though https://github.com/grafana/grafana/pull/9208 would fix the conflict issues

**Comment by mtanda** [comment]
No error output. Just show metric name in label matcher
image

**Comment by torkelo** [comment]
ok, good. that is as I expected. will it be fixed by  #9208 ?

**Comment by mtanda** [comment]
Yes, it fix the problem.

**Pull request merged**